cmd/docker/docker.go

```diff
@@ -308,9 +308,11 @@ func runDocker(ctx context.Context, dockerCli *command.DockerCli) error {
 		return err
 	}
 
-	mp := dockerCli.MeterProvider(ctx)
-	defer mp.Shutdown(ctx)
-	otel.SetMeterProvider(mp)
+	mp := dockerCli.MeterProvider()
+	if mp, ok := mp.(command.MeterProvider); ok {
+		defer mp.Shutdown(ctx)
+	}
```

Comment:
Since we don't expect the typecast to actually fail, I think we should have an error log (to stderr) just in case any future OTEL update breaks our assumption that these methods are defined.

Reply:
@krissetto can you TAL at that real quick?